tests/imageuploadcommand.js

```diff
@@ -78,6 +79,21 @@ describe( 'ImageUploadCommand', () => {
 			expect( getModelData( document ) ).to.equal( `<image src="image.png"></image>[<image uploadId="${ id }"></image>]` );
 		} );
 
+		it( 'should allow inserting image at some custom position', () => {
+			const file = createNativeFileMock();
+			setModelData( document, '[<paragraph>foo</paragraph>]<paragraph>bar</paragraph>' );
```

Comment:
This isn't correct model selection. Selection should not contain a paragraph from outside but we inside it.